sysinfogen.sh

```diff
@@ -238,7 +238,7 @@ cat >> "$OUTFILE" << EOF
 #define SYSINFO_CPU "$( cleanstring "${HERC_CPU}" )"
 
 // CPU Cores (Platform-dependent)
-#define SYSINFO_CPUCORES ( $( cleanstring "${HERC_CORES}" ) )
+#define SYSINFO_CPUCORES ( "$( cleanstring "${HERC_CORES}" )" )
```

Comment:
This should evaluate to an int rather than a string (it gets assigned to an int variable in [src/common/sysinfo.c](https://github.com/HerculesWS/Hercules/blob/b5a59b17f29a1e36ce815925218e8b9c96ed8f2b/src/common/sysinfo.c#L53))

I think keeping it as an integer has some value, since it can be used in plugins or custom code to take decisions based on the number of cores available.

What do you think about something like this instead? (in your case it'd output `( 10 )`)

```diff --git i/sysinfogen.sh w/sysinfogen.sh
index f96a1c492..fa5bcc3cf 100755
--- i/sysinfogen.sh
+++ w/sysinfogen.sh
@@ -190,7 +190,7 @@ case $HERC_PLATFORM in
 			HWDATA="$( system_profiler SPHardwareDataType )"
 			HWDATA_CPU="$( echo "$HWDATA" | grep "Processor Name:" | cut -d: -f2- )"
 			HWDATA_CPUSPEED="$( cleanstring "$( echo "$HWDATA" | grep "Processor Speed:" | cut -d: -f2- )" )"
-			HERC_CORES="$( echo "$HWDATA" | grep "Total Number of Cores:" | cut -d: -f2- )"
+			HERC_CORES="$( echo "$HWDATA" | grep "Total Number of Cores:" | cut -d: -f2- | sed -E 's/ *([0-9]+).*/\1/')"
 			HERC_CPU="${HWDATA_CPU} (${HWDATA_CPUSPEED})"
 		fi
 		;;

```

I'm not sure if it's worth also exposing the details of how many performance and efficiency cores are available (and I'd need to research how to do so on other OSes as well), if we want to add those as well to the sysinfo information, but I'd still keep those as separate integer values, not as a string